### PR TITLE
DEV: change io loader classes app_type to LOADER

### PR DESCRIPTION
**Pull request by Nick-Foto**

### CODE REVIEW

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
small edit.

Plus, there are 6 tests marked as `xfail` that start with `test_load_..`. Do these still fail?

**Comment by Nick-Foto** [comment]
> small edit.
> 
> Plus, there are 6 tests marked as `xfail` that start with `test_load_..`. Do these still fail?

`test_load_db_failure_json_file` fails with this error:
>           writer.data_store.db.close()
E           AttributeError: 'write_db' object has no attribute 'data_store'

**Comment by Nick-Foto** [comment]
> small edit.
> 
> Plus, there are 6 tests marked as `xfail` that start with `test_load_..`. Do these still fail?

`test_load_tabular_distance_matrix` failed with this error:

output = data.to_string(format=self._format)
>       return self.data_store.write(identifier, output)
E       AttributeError: 'write_tabular' object has no attribute 'data_store'

File: src/cogent3/app/composable.py [hunk 1]
```diff
@@ -1092,8 +1092,8 @@ def _apply_to(
         logger.log_versions(["cogent3"])
 
     results = []
-    process = self.input or self
-    if self.input:
+    process = self.input or self if self.app_type != LOADER else self
+    if self.app_type != LOADER and self.input:
         # As we will be explicitly calling the input object, we disconnect
         # the two-way interaction between input and self. This means self
         # is not called twice, and self is not unecessarily pickled during
```
[the remark this answers is not in the document]
Comment (Nick-Foto) [reply src/cogent3/app/composable.py:1099]: `test_load_tabular_motif_counts_array` failed with this error :

output = data.to_string(format=self._format)
>       return self.data_store.write(identifier, output)
E       AttributeError: 'write_tabular' object has no attribute 'data_store'

[the remark this answers is not in the document]
Comment (Nick-Foto) [reply src/cogent3/app/composable.py:1099]: `test_load_tabular_motif_freqs_array` failes with this error:


output = data.to_string(format=self._format)
>       return self.data_store.write(identifier, output)
E       AttributeError: 'write_tabular' object has no attribute 'data_store'

[the remark this answers is not in the document]
Comment (Nick-Foto) [reply src/cogent3/app/composable.py:1099]: `test_load_tabular_pssm` fails with this error:

output = data.to_string(format=self._format)
>       return self.data_store.write(identifier, output)
E       AttributeError: 'write_tabular' object has no attribute 'data_store'

[the remark this answers is not in the document]
Comment (Nick-Foto) [reply src/cogent3/app/composable.py:1099]: `test_load_tabular_table` failed with this error:

output = data.to_string(format=self._format)
>       return self.data_store.write(identifier, output)
E       AttributeError: 'write_tabular' object has no attribute 'data_store'

[the remark this answers is not in the document]
Comment (Nick-Foto) [reply src/cogent3/app/composable.py:1099]: All 6 tests failed.

**Pull request merged**